pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -159,6 +159,18 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 		return fmt.Errorf("failed to get imageStream %s from cluster %s: %w", isName.String(), srcClusterName, err)
 	}
 
+	if sourceImageStream.ObjectMeta.DeletionTimestamp.IsZero() {
+		if err := ensureFinalizer(ctx, sourceImageStream, r.registryClients[srcClusterName]); err != nil {
+			return fmt.Errorf("failed to ensure finalizer to %s from cluster %s: %w", isName.String(), srcClusterName, err)
+		}
+	} else {
+		if err := finalizeIfNeeded(ctx, sourceImageStream, srcClusterName, r.registryClients); err != nil {
```

Comment:
This code can be more left-aligned and avoid the else by doing:

```
if sourceImageStream.DeletionTimestamp != nil {
  if err := finalizeIfNeeded(); err != nil {
    return fmt.Errorf("cleanup failed") 
  }
  return nil
}

if err := ensureFinalizer(); err != nil {
  return fmt.Errorf("failed to ensure finalizer")
}


/// Rest  of reconciler
```

Reply:
Updated.